src/services/auth.service.js

```diff
@@ -1,3 +1,42 @@
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcrypt');
 
+const userDao = require('../models/user.dao');
+const { validateEmail } = require('../utils/validators');
 
+const signUp = async (email, password) => {
+  validateEmail(email);
 
+  const user = await userDao.getUserByEmail(email);
+
+  if (user) {
+    const err = new Error('duplicated email');
+    err.statusCode = 400;
+    throw err;
+  }
+
+  const hashedPassword = await bcrypt.hash(password, 10);
+  await userDao.createUser(email, hashedPassword);
+};
+
+const signIn = async (email, password) => {
+  const user = await userDao.getUserByEmail(email);
+
+  if (!user) {
+    const err = new Error('specified user does not exist');
+    err.statusCode = 404;
+    throw err;
+  }
+
+  const result = await bcrypt.compare(password, user.password);
+
+  if (!result) {
+    const err = new Error('invalid password');
+    err.statusCode = 401;
+    throw err;
+  }
+
+  return jwt.sign({ sub: user.id}, process.env.JWT_SECRET);
```

Comment:
jwt 공식 홈페이지에서 확인하면 `sub` 는 subject로써 토큰 제목을 가리킵니다.

payload로 user의 정보를 등록하고 싶다면 다음과 같은 방법을 사용할 수 있습니다.
- userId 라는 클레임을 새로 등록하기
- aud 클레임에 `user.id` 등록하기

```suggestion
  return jwt.sign({ userId: user.id}, process.env.JWT_SECRET);
```